lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,412 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: redhat_repositories
+short_description: Manage repositories with RHSM using the C(subscription-manager) command
+description:
+  - Manage repositories with the Red Hat Subscription Management entitlement
+    platform using the C(subscription-manager) command
+author: "Jerome Fenal (@jfenal)"
+notes:
+  - In order to be able to enable repositories, a system will
+    need first to be subscribed to RHSM. Use the
+    C(redhat_subscription) Ansible module for that matter.
+version_added: "2.4"
+requirements:
+  - subscription-manager
+options:
+  id:
+    description:
+      - id of repositories to enable or disable.
+        When specifying multiple repos, separate them with a C(,).
+        To specify all repositories, use C(*).
+        In incremental mode, other repos won't be touched.
+        In idempotent mode (default), other repos would be disabled or enabled
+        depending on their actual state.
+    required: False
+    default: null
+  list:
+    description:
+      - List either all repositories, or only enabled or disabled.
+    required: False
+    default: all
+    choices: [ "all", "enabled", "disabled" ]
+  mode:
+    description:
+      - Use either the idempotent or the incremental mode.
+    required: False
+    default: "idempotent"
+    choices: [ "idempotent", "incremental" ]
+  state:
+    description:
+      - Whether to enable (C(enabled)) or disable (C(disabled)) a repository.
+    required: False
+    choices: [ "enabled", "disabled" ]
+    default: "enabled"
+
+'''
+
+EXAMPLES = '''
+# List all repositories
+- redhat_repositories:
+    list: all
+
+# List enabled repositories
+- redhat_repositories:
+    list: enabled
+
+# List disabled repositories
+- redhat_repositories:
+    list: disabled
+
+# Enable only those repositories (implicit default idempotent mode)
+- redhat_repositories:
+    id:
+      - rhel-7-server-rpms
+      - rhel-7-server-optional-rpms
+    state: enabled
+
+# Enable only those repositories (explicit idempotent mode)
+- redhat_repositories:
+    id:
+      - rhel-7-server-rpms
+      - rhel-7-server-optional-rpms
+    state: enabled
+    mode: idempotent
+
+# Disable repositories
+- redhat_repositories:
+    id:
+      - .*-beta-.*
+      - .*-htb-.*
+      - .*-eus-.*
+      - .*-aus-.*
+    state: disabled
+    mode: incremental
+
+# Disable all repositories (mode=incremental must be specified for now)
+- redhat_repositories:
+    id: "*"
+    state: disabled
+    mode: incremental
+'''
+
+RETURN = '''
+redhat_repositories:
+    description: list of repositories and their data
```

Comment:
Please change the indentation to 2 characters to have the same formatting like above.